src/feature/client/transports.c

```diff
@@ -1902,3 +1933,46 @@ managed_proxy_severity_parse(const char *severity)
 
   return -1;
 }
+
+/** Return the outbound address from the given <b>family</b>. Returns NULL if
+ * the user haven't specified a specific outbound address in either
+ * OutboundBindAddress or OutboundBindAddressPT. */
+STATIC const tor_addr_t *
+managed_proxy_outbound_address(const or_options_t *options, sa_family_t family)
+{
+  tor_assert(options);
+
+  const tor_addr_t *address = NULL;
+  int family_index;
+
+  switch (family) {
+  case AF_INET:
+    family_index = 0;
+    break;
+  case AF_INET6:
+    family_index = 1;
+    break;
+  default:
+    /* LCOV_EXCL_START */
+    tor_assert_unreached();
+    return NULL;
+    /* LCOV_EXCL_STOP */
+  }
+
+  /* We start by checking if the user specified an address in
```

Comment:
There are now three copies of this code in tor, the other two copies are in conn_get_outbound_address().
This implementation is different to the other copies.

Let's refactor this code into a new function, maybe in config.c?

A nice interface could look like:
STATIC const tor_addr_t *
config_get_outbound_address(const or_options_t *options, sa_family_t family, outbound_addr_t type)

And it would also solve the other code duplication issue.